backend/module/chaos/serverexperimentation/xds/ecds_faults_generator.go

```diff
@@ -172,69 +114,57 @@ func generateECDSResource(experiments []*experimentstore.Experiment, cluster str
 		}
 	}
 
-	logger.Debugw("ECDS Fault filter", "faultFilter", faultFilter, "upstreamcluster", upstreamCluster)
-
 	serializedFaultFilter, err := proto.Marshal(faultFilter)
 	if err != nil {
-		logger.Warnw("Unable to unmarshal fault filter", "faultFilter", faultFilter)
-		return []gcpTypes.ResourceWithTtl{}
+		return nil, err
 	}
 
-	return []gcpTypes.ResourceWithTtl{{
-		Resource: &gcpCoreV3.TypedExtensionConfig{
-			Name: faultFilterName,
-			TypedConfig: &any.Any{
-				TypeUrl: faultFilterTypeURL,
-				Value:   serializedFaultFilter,
-			},
+	config := &gcpCoreV3.TypedExtensionConfig{
+		Name: faultFilterName,
+		TypedConfig: &any.Any{
+			TypeUrl: faultFilterTypeURL,
+			Value:   serializedFaultFilter,
 		},
-		Ttl: ttl,
-	}}
+	}
+
+	return xds.NewECDSResource(enforcingCluster, config)
 }
 
-func generateEmptyECDSResource(cluster string, ecdsConfig *ECDSConfig, logger *zap.SugaredLogger) []gcpTypes.ResourceWithTtl {
-	// cache serialized
-	if SerializedDefaultFaultFilter == nil {
+func (g *ECDSFaultsGenerator) GenerateDefaultResource(cluster string, resourceName string) (*xds.ECDSResource, error) {
+	if resourceName != faultFilterConfigNameForIngressFault && !strings.HasPrefix(resourceName, fmt.Sprintf(faultFilterConfigNameForEgressFault, "")) {
+		return xds.NewEmptyECDSResource(), nil
+	}
+
+	if g.SerializedDefaultFaultFilter == nil {
```

Comment:
should this be done in some initializer instead? seems odd to have it happen inline here the first time we call this

Reply:
good point